corehq/apps/data_analytics/tests/test_malt_generator.py

```diff
@@ -0,0 +1,134 @@
+import datetime
+from django.test import TestCase
+
+from corehq.apps.app_manager.const import APP_V2, AMPLIFIES_YES
+from corehq.apps.app_manager.models import Application
+from corehq.apps.data_analytics.malt_generator import MALTTableGenerator
+from corehq.apps.data_analytics.models import MALTRow
+from corehq.apps.domain.models import Domain
+from corehq.apps.users.models import CommCareUser
+from corehq.apps.smsforms.app import COMMCONNECT_DEVICE_ID
+from corehq.apps.sofabed.models import FormData, MISSING_APP_ID
+
+from dimagi.utils.dates import DateSpan
+
+
+class MaltGeneratorTest(TestCase):
+
+    DOMAIN_NAME = "test"
+    USERNAME = "malt-user"
+    DEVICE_ID = "my_phone"
+    UNKNOWN_ID = "UNKNOWN_ID"
+
+    correct_date = datetime.datetime.now()
+    out_of_range_date = correct_date - datetime.timedelta(days=32)
+    malt_month = DateSpan.from_month(correct_date.month, correct_date.year)
+
+    @classmethod
+    def setUpClass(cls):
+        cls._setup_domain_user()
+        cls._setup_apps()
+        cls._setup_sofabed_forms()
+        cls.run_malt_generation()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain.delete()
+        FormData.objects.all().delete()
+        MALTRow.objects.all().delete()
+
+    @classmethod
+    def _setup_domain_user(cls):
+        cls.domain = Domain(name=cls.DOMAIN_NAME)
+        cls.domain.save()
+        cls.user = CommCareUser.create(cls.DOMAIN_NAME, cls.USERNAME, '*****')
+        cls.user.save()
+        cls.user_id = cls.user._id
+
+    @classmethod
+    def _setup_apps(cls):
+        cls.app = Application.new_app(cls.DOMAIN_NAME, "app 1", APP_V2)
+        cls.wam_app = Application.new_app(cls.DOMAIN_NAME, "app 2", APP_V2)
+        cls.wam_app.amplifies_workers = AMPLIFIES_YES
+        cls.app.save()
+        cls.wam_app.save()
+        cls.app_id = cls.app._id
+        cls.wam_app_id = cls.wam_app._id
+
+    @classmethod
+    def _setup_sofabed_forms(cls):
+        form_data_rows = []
+        common_args = {  # values don't matter
+            'time_start': cls.correct_date,
+            'time_end': cls.correct_date,
+            'duration': 10,
+        }
+
+        def _form_data(instance_id,
+                       app_id,
+                       domain=cls.DOMAIN_NAME,
+                       received_on=cls.correct_date,
+                       device_id=cls.DEVICE_ID,
+                       user_id=cls.user_id):
```

Comment:
Seems you could get rid of the `domain` and `user_id` params since they are never passed in.